lib/guardian/plug/ensure_permissions.ex

```diff
@@ -63,19 +71,38 @@ defmodule Guardian.Plug.EnsurePermissions do
     case Guardian.Plug.claims(conn, key) do
       {:ok, claims} ->
         perms = Map.get(opts, :perms, %{})
-        result = Enum.all?(Map.get(opts, :perm_keys), fn(perm_key) ->
-          found_perms = Guardian.Permissions.from_claims(claims, perm_key)
-          Guardian.Permissions.all?(
-            found_perms,
-            Map.get(perms, perm_key),
-            perm_key
-          )
-        end)
+        result =
```

Comment:
I like this approach :) but when does `matches_permissions?(_, _, [])` get its pattern matched? Also, should it not be private?
Another note is below regarding the use of Enum.any.